service/matching/matching_engine_test.go

```diff
@@ -2747,6 +2766,376 @@ func (s *matchingEngineSuite) TestUpdatePhysicalTaskQueueGauge_BuildID() {
+	var donePolling bool
+	for !donePolling {
+		result, err := s.matchingEngine.PollWorkflowTaskQueue(context.Background(), &matchingservice.PollWorkflowTaskQueueRequest{
+			NamespaceId: namespaceId,
+			PollRequest: &workflowservice.PollWorkflowTaskQueueRequest{
+				TaskQueue: taskQueue,
+				Identity:  "nobody",
+			},
+		}, metrics.NoopMetricsHandler)
+		if len(result.TaskToken) == 0 || result.GetAttempt() == 0 {
+			continue
+		}
+		if err != nil {
+			// DB could have failed while fetching tasks; try again
+			continue
+		}
+		s.NoError(err)
+		donePolling = true
+	}
+}
+
+// addWorkflowTasks is a helper that adds taskCount number of tasks for each numWorker
+func (s *matchingEngineSuite) addWorkflowTasks(concurrently bool, numWorkers int, taskCount int,
+	taskQueue *taskqueuepb.TaskQueue, workflowExecution *commonpb.WorkflowExecution, wg *sync.WaitGroup) {
+	if concurrently {
+		for p := 0; p < numWorkers; p++ {
+			go func() {
+				for i := 0; i < taskCount; i++ {
+					s.addWorkflowTask(workflowExecution, taskQueue)
+				}
+				wg.Done()
+			}()
+		}
+	} else {
+		// Add tasks sequentially
+		for p := 0; p < numWorkers; p++ {
+			for i := 0; i < taskCount; i++ {
+				s.addWorkflowTask(workflowExecution, taskQueue)
+			}
+		}
+	}
+}
+
+// pollWorkflowTasks polls tasks using numWorkers
+func (s *matchingEngineSuite) pollWorkflowTasks(concurrently bool, workflowType *commonpb.WorkflowType, numPollers int, taskCount int,
+	ptq *PhysicalTaskQueueKey, taskQueue *taskqueuepb.TaskQueue, wg *sync.WaitGroup) {
+	s.mockHistoryWhilePolling(workflowType)
+	if concurrently {
+		for p := 0; p < numPollers; p++ {
+			go func() {
+				for i := 0; i < taskCount; i++ {
+					s.createPollWorkflowTaskRequestAndPoll(taskQueue)
+				}
+				wg.Done()
+			}()
+		}
+	} else {
+		tasksPolled := 0
+		for i := 0; i < numPollers*taskCount; i++ {
+			s.createPollWorkflowTaskRequestAndPoll(taskQueue)
+			tasksPolled += 1
+
+			// PartitionManager could have been unloaded; fetch the latest copy
+			pgMgr := s.getPhysicalTaskQueueManagerImpl(ptq)
+			s.LessOrEqual(int64(taskCount-tasksPolled), pgMgr.backlogMgr.db.getApproximateBacklogCount())
+		}
+	}
+}
+
+// getPhysicalTaskQueueManagerImpl extracts the physicalTaskQueueManagerImpl for the given PhysicalTaskQueueKey
+func (s *matchingEngineSuite) getPhysicalTaskQueueManagerImpl(ptq *PhysicalTaskQueueKey) *physicalTaskQueueManagerImpl {
+	pgMgr, ok := s.matchingEngine.partitions[ptq.Partition().Key()].(*taskQueuePartitionManagerImpl).defaultQueue.(*physicalTaskQueueManagerImpl)
+	s.True(ok, "taskQueueManger doesn't implement taskQueuePartitionManager interface")
+	return pgMgr
+}
+
+func (s *matchingEngineSuite) addConsumeAllWorkflowTasksNonConcurrently(taskCount int, numWorkers int, numPollers int) {
+	s.matchingEngine.config.LongPollExpirationInterval = dynamicconfig.GetDurationPropertyFnFilteredByTaskQueue(10 * time.Millisecond)
```

Comment:
this is overridden by default to 100ms.. does 10ms make a difference for this test?

Reply:
removed, thanks!